src/Interpreters/Cache/FileSegment.cpp

```diff
@@ -813,6 +813,7 @@ void FileSegment::complete(bool allow_background_download)
                     {
                         cache_writer->finalize();
                         cache_writer.reset();
+                        remote_file_reader.reset();
```

Comment:
not sure if we want to do it for `PARTIALLY_DOWNLOADED_NO_CONTINUATION` below and in `setDownloadFailedUnlocked`
just to be safe